src/main_app/jobs_workers/crop_main_files/process_new.py

```
@@ -0,0 +1,456 @@
+        file_info.steps[step] = {"result": False, "msg": error}
+        file_info.status = "failed"
+        file_info.error = error
+        self.result["summary"]["failed"] += 1
+
+    def _skip_step(self, file_info: FileProcessingInfo, step: str, reason: str) -> None:
+        """Mark a step as skipped (result=None)."""
+        file_info.steps[step] = {"result": None, "msg": reason}
+
+    def _skip_upload_steps(self, file_info: FileProcessingInfo) -> None:
+        for step in ("upload_cropped", "update_original", "update_template"):
+            self._skip_step(file_info, step, "Skipped – upload disabled")
```

Ruff (0.15.2)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)